Code/Framework/AzCore/AzCore/std/iterator/unreachable_sentinel.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/base.h>
+#include <AzCore/std/concepts/concepts.h>
+
+
+namespace AZStd
+{
+    //! Represents the upper bound of an unbounded range
+    //! It is never equal to the iterator being compared against.
+    struct unreachable_sentinel_t
```

Comment:
This is really cool, is this so you can essentially have infinite lazy ranges?

Reply:
Yes, the `AZstd::unreachable_sentinel` can be used with the std::range:: algorithms to loop until a condition is met.
It is what ranges::iota_view and ranges::repeat_view use to iterate until a loop is broken out of.

Alternatively it could be used to find the string length of a null-terminated string using AZStd::ranges::find(not an efficient way to get the string length, but a viable alternative).

Reply:
Interesting! 😄 Thanks for the explanation